### PR TITLE
notes e2e test

### PR DESCRIPTION
**Pull request by rajeshg**
Add e2e test for notes basic features
## Test Plan

Run playwright tests
## Checklist

- [x] Tests updated
- [x] Docs updated

### CODE REVIEW

File: tests/e2e/notes.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { faker } from '@faker-js/faker'
+import { expect, test } from '../playwright-utils.ts'
+
+test('Users can create notes', async ({ login, page }) => {
+	const user = await login()
+	await page.goto(`/users/${user.username}/notes`)
+
+	const newNote = createNewNote()
+	await page.locator("//a[text()='+ New Note']").click()
```
Comment (kentcdodds) [inline tests/e2e/notes.test.ts:9]: Let's swap from `.locator` to the testing-library inspired API playwright added a while ago. Check the recommended locators here: https://playwright.dev/docs/locators#quick-guide

File: tests/e2e/notes.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	// blank form submission should result in errors
+	await page.locator('button[type="submit"]').click()
+	await expect(page.locator('ul#note-editor-title-error')).toBeVisible()
+	await expect(page.locator('ul#note-editor-content-error')).toBeVisible()
+
+	// fill in form and submit
+	await page.locator('input[name="title"]').fill(newNote.title)
+	await page.locator('textarea[name="content"]').fill(newNote.content)
+	await page.locator('button[type="submit"]').click()
+	await expect(page).toHaveURL(new RegExp(`/users/${user.username}/notes/*`))
```
Comment (kentcdodds) [inline tests/e2e/notes.test.ts:20]: Should this be:

```suggestion
	await expect(page).toHaveURL(new RegExp(`/users/${user.username}/notes/.*`))
```

Same thing below

File: tests/e2e/notes.test.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+})
+
+test('Users can edit notes', async ({ login, page }) => {
+	const user = await login()
+	await page.goto(`/users/${user.username}/notes`)
```
Comment (kentcdodds) [inline tests/e2e/notes.test.ts:25]: I noticed the two tabs here. I suggest you install a prettier plugin for your editor so you get format on save. It saves a lot of time!

**Review by kentcdodds — CHANGES_REQUESTED** [review]
This is very cool! Thanks for working on this.

I'm thinking we may want to remove this when people generate the project because while it's nice for us to have to verify we don't break stuff, it's just another thing for people to delete when they start building their app.

File: tests/e2e/notes.test.ts (hunk 1 above)
Comment (rajeshg) [reply tests/e2e/notes.test.ts]: Thanks! Removed pretty much all `.locator` usage, except a couple of instances where I needed to verify if deletion works as intended.

**Comment by kentcdodds** [comment]
It looks like there's a lot more unnecessary use of `.locator`.

**Review by kentcdodds — APPROVED** [review]
Looks great! Thank you 👏

**Pull request merged**